enhancements/kata-containers/kata-containers-dev-preview.md

```diff
@@ -0,0 +1,259 @@
+
+
+## Goals
+- Provide a way to enable running user workloads on kata containers on an OpenShift cluster.
+- Clusters with kata container workloads should support upgrades.
+- Add security policy to control which users can run what workloads on kata in an OpenShift cluster.
+
+
+## Non-goals
+- Although kata containers are capable of supporting running a different kernel then the one used on the Openshift node, such features will not be supported. This is in order to simplify the development cycles and testing efforts of this project.
+- Running OpenShift control plane and core operators on kata. These will continue to run on runc.
+
+
+## Kata containers architecture
+
+For details on the kata architecture see https://github.com/kata-containers/documentation/blob/master/design/architecture.md
+
+## Proposal - main development efforts
+### Overview
+Kubernetes provides support for RuntimeClasses. RuntimeClass is a feature for selecting the container runtime configuration. The container runtime configuration is used to run a Pod’s containers.
+
+CRI-O today comes out of the box with a runc as the default runtime. CRI-O also supports RUntimeClasses and using this configuration, it will support a KataContainers runtime as well.
+
+The 2 fundamental problems we need to address for getting kata to work with Openshift are:
+
+1. Getting the kata artifacts on to an RHCOS node
+2. Deploying kata and configuring CRI-O to use it as the runtime
+
+For addressing the first issue of packaging and installing kata in openshift we will introduce 2 approaches, a short term and a long term.
+
+For addressing the second issue of deploying and configuring CRI-O we will use an operator.
+
+The next sections detailed the 2 solutions.
+
+### KataContainer Packaging
+#### Short term (preview release): Use container images to deliver RPMs and install using rpm-ostree
+In this option we build a container image containing RHEL AV RPMs (such as QEMU) and kata RPMs. The kata operator will create a pod based on this image and then mount the host filesystem and install the RPMs on the filesystem (kata operators will be described in a later section).
+
+**Pros:**
+1. The RPMs can be installed only by the users who would be interested in using the kata runtime
+2. There's no extra work related to properly loading qemu-kiwi dependencies, as all the packages would be in installed in the same PATH its build was intent for
+3. QEMU RPMs have been tested / validated using an identical environment
+4. No additional work with regarding to re-packaging those in another form other than the RPMs which is the way they are consumed by other Layered Products
+
+**Cons:**
+1. ~200 mb extra, installed in the host, for those who'd be using kata runtime
+2. Updates / Removal may be more complicated than having the RPMs as part of machine-os-content
+
+#### Long term: Use RHCOS extensions (qemu-kiwi and dependencies only)
```

Comment:
This section looks like a copy/paste from https://github.com/openshift/enhancements/pull/317

Could this be re-written to just link to that enhancement and instead focus this section on what packages would be supplied in the extension for kata containers?

Reply:
Please update that section to indicate what we would put in the extension.

Do we want to discuss the reduced qemu package here?

Reply:
changed it, thank you!